elements/rendering/utils/program_loader.cpp

```diff
@@ -44,8 +44,47 @@ bool program_data::read(const pugi::xml_document & doc)
     if(fragment_node.text().empty())
         return false;
 
-    v_shader_ = vertex_node.text().get();
-    f_shader_ = fragment_node.text().get();
+    const auto vertex_shader_length = std::strlen(vertex_node.text().get());
+    const auto fragment_shader_length = std::strlen(fragment_node.text().get());
+
+    v_shader_.reserve(18 + vertex_shader_length);
+    f_shader_.reserve(18 + fragment_shader_length);
+    v_shader_.clear();
+    f_shader_.clear();
+
+    const auto shaders_version = root_node.child("shaders").attribute("version");
+    if (!shaders_version.empty())
+    {
+        const auto shaders_version_str = shaders_version.as_string();
```

Comment:
const char * is more readable. Versions are small strings, we can use std::string with SSO.